pkg/util/pki/subject_test.go

```diff
@@ -112,5 +113,94 @@ func TestShouldFailForHexDER(t *testing.T) {
 		t.Fatal("expected error, but got none")
 	}
 
-	assert.Contains(t, err.Error(), "failed to unmarshal hex-encoded string: asn1: syntax error: data truncated")
+	assert.Contains(t, err.Error(), "failed to decode BER encoding: unexpected EOF")
+}
+
+func TestRoundTripRDNSequence(t *testing.T) {
+	rdnSequences := []pkix.RDNSequence{
+		{
+			[]pkix.AttributeTypeAndValue{
+				{Type: OIDConstants.Organization, Value: "Corp."},
+				{Type: OIDConstants.OrganizationalUnit, Value: "FooLong"},
+			},
+		},
+		{
+			[]pkix.AttributeTypeAndValue{
+				{Type: OIDConstants.CommonName, Value: "foo-lon❤️\\g.com    "},
+				{Type: OIDConstants.OrganizationalUnit, Value: "Foo===Long"},
+				{Type: OIDConstants.OrganizationalUnit, Value: "Ba  rq"},
+				{Type: OIDConstants.OrganizationalUnit, Value: "Baz"},
+			},
+			[]pkix.AttributeTypeAndValue{
+				{Type: OIDConstants.Organization, Value: "C; orp."},
+				{Type: OIDConstants.Country, Value: "US"},
+			},
+		},
+		{
+			[]pkix.AttributeTypeAndValue{
+				{Type: OIDConstants.CommonName, Value: "fo\x00o-long.com"},
+			},
+		},
+	}
+
+	for _, rdnSeq := range rdnSequences {
+		subjectString := rdnSeq.String()
+
+		newRDNSeq, err := UnmarshalSubjectStringToRDNSequence(subjectString)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		ans1Seq1, err := asn1.Marshal(rdnSeq)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		ans1Seq2, err := asn1.Marshal(newRDNSeq)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		assert.Equal(t, ans1Seq1, ans1Seq2)
+	}
+}
+
+func FuzzRoundTripRDNSequence(f *testing.F) {
+	f.Add("CN=foo-long.com,OU=FooLong,OU=Barq,OU=Baz,OU=Dept.,O=Corp.,C=US")
+	f.Add("CN=foo-lon❤️\\,g.com,OU=Foo===Long,OU=Ba # rq,OU=Baz,O=C\\; orp.,C=US")
+	f.Add("CN=fo\x00o-long.com,OU=\x04FooLong")
```

Comment:
Why are these particular samples added to the fuzz test? They've already been tested in `TestRoundTripRDNSequence` above, right?

Add a comment to explain. Or remove.

Reply:
True, I think that these examples are actually run as tests when you run go test without fuzz.
So this Fuzz test is a replacement for the Test function.

Reply:
I removed the Test below and kept only this fuzz function.